Makefile

```diff
@@ -178,7 +178,7 @@ test: manifests generate fmt lint test-unit test-e2e #HELP Run all tests.
 
 .PHONY: e2e
 e2e: #EXHELP Run the e2e tests.
-	go test -count=1 -v ./test/e2e/...
+	go test -count=1 -v -run "$(if $(TEST_FILTER),$(TEST_FILTER),.)" ./test/e2e/...
```

Comment:
Why we need TEST_FILTER?

Reply:
+1, I am also curious about why we need TEST_FILTER?

Reply:
I thought about leaving it out, but it was very handy for repeatedly running only one of the e2e test while developing.

Reply:
IHMO, that adds more complexities, could we remove? That is not clear when/how to use
If we want to run only 1 test, we can use go test manually and pass the test right?

Reply:
Personally, I'm a fan of it. What's the worry in terms of complexity? It stays mostly hidden anyway...?
If anything I'd update the help to call it out XDD but I'm easy either way.

Reply:
seem okay to keep it? I've been using it like:

```bash
make test-e2e TEST_FILTER="Test.*MetricsExportedEndpoint"
```

Reply:
I like it too! Could we make sure that the unit test recipe uses this same variable?

Reply:
@perdasilva having trouble seeing where to add for docs, do we mention make targets in help now?

Reply:
@joelanford to be equivalent for test unit we need to change like:
```bash
make test-unit TEST_PKG=./internal/operator-controller/applier TEST_FILTER=TestApply_Installation
```

I am committing this change now.